packages/xstate-react/src/useMachine.ts

```diff
@@ -0,0 +1,241 @@
+import { useState, useEffect, useRef, useLayoutEffect } from 'react';
+import {
+  interpret,
+  EventObject,
+  StateMachine,
+  State,
+  Interpreter,
+  InterpreterOptions,
+  MachineOptions,
+  StateConfig,
+  Typestate,
+  ActionObject,
+  ActionFunction,
+  ActionMeta
+} from 'xstate';
+import useConstant from './useConstant';
+import { partition } from './utils';
+
+enum ReactEffectType {
+  Effect = 1,
+  LayoutEffect = 2
+}
+
+export interface ReactActionFunction<TContext, TEvent extends EventObject> {
+  (
+    context: TContext,
+    event: TEvent,
+    meta: ActionMeta<TContext, TEvent>
+  ): () => void;
+  __effect: ReactEffectType;
+}
+
+export interface ReactActionObject<TContext, TEvent extends EventObject>
+  extends ActionObject<TContext, TEvent> {
+  exec: ReactActionFunction<TContext, TEvent>;
+}
+
+function createReactActionFunction<TContext, TEvent extends EventObject>(
+  exec: ActionFunction<TContext, TEvent>,
+  tag: ReactEffectType
+): ReactActionFunction<TContext, TEvent> {
+  const effectExec: unknown = (...args: Parameters<typeof exec>) => {
+    // don't execute; just return
+    return () => {
+      return exec(...args);
+    };
+  };
+
+  Object.defineProperties(effectExec, {
+    name: { value: `effect:${exec.name}` },
+    __effect: { value: tag }
+  });
+
+  return effectExec as ReactActionFunction<TContext, TEvent>;
+}
+
+export function asEffect<TContext, TEvent extends EventObject>(
+  exec: ActionFunction<TContext, TEvent>
+): ReactActionFunction<TContext, TEvent> {
+  return createReactActionFunction(exec, ReactEffectType.Effect);
```

Comment:
FYI @davidkpiano Wanted to build latest master to get these changes to my project and TS is complaining here

![image](https://user-images.githubusercontent.com/1096340/83961439-56e92400-a893-11ea-9f3c-3426a19bacd3.png)

Reply:
Are you using the latest TypeScript? I do not get this error.

Reply:
There is TS `3.8.3` in yarn.lock, so that's not the latest.